AAUnlimited/MemMods/AAPlay/Events/NpcActions.cpp

```diff
@@ -847,20 +849,37 @@ void headTrackingChangeInjection() {
 
 void __stdcall extraHairFix(DWORD* charAddress, BYTE value) {
 	extraHairTest = value;
-	int i = 0;
-	while (Shared::GameState::getConversationCharacter(i)) {
-		auto card1 = Shared::GameState::getConversationCharacter(i);
-		if (card1->m_xxSkeleton) {
-			DWORD* somepointer = (DWORD*)((char*)(card1->m_xxSkeleton->m_unknown13) + 0x88);
-			if (charAddress == somepointer) {
-				auto hairs = AAPlay::g_characters[card1->m_seat].m_cardData;
-				if (hairs.GetHairs(0).size() || hairs.GetHairs(1).size() || hairs.GetHairs(2).size() || hairs.GetHairs(3).size()) {
-					extraHairTest = 1;
-					return;
+	if (General::IsAAPlay) {
+		int i = 0;
+		while (Shared::GameState::getConversationCharacter(i)) {
+			auto card1 = Shared::GameState::getConversationCharacter(i);
+			if (card1->m_xxSkeleton) {
+				DWORD* somepointer = (DWORD*)((char*)(card1->m_xxSkeleton->m_unknown13) + 0x88);
+				if (charAddress == somepointer) {
+					auto hairs = AAPlay::g_characters[card1->m_seat].m_cardData;
+					if (hairs.GetHairs(0).size() || hairs.GetHairs(1).size() || hairs.GetHairs(2).size() || hairs.GetHairs(3).size()) {
+						extraHairTest = 1;
+						return;
+					}
+				}
+			}
+			i++;
+		}
+	}
+	else if (General::IsAAEdit) {
+		CharInstData* card1 = &AAEdit::g_currChar;
+		if (card1->m_char) {
+			if (card1->m_char->m_xxSkeleton) {
+				DWORD* somepointer = (DWORD*)((char*)(card1->m_char->m_xxSkeleton->m_unknown13) + 0x88);
+				if (charAddress == somepointer) {
+					auto hairs = card1->m_cardData;
+					if (hairs.GetHairs(0).size() || hairs.GetHairs(1).size() || hairs.GetHairs(2).size() || hairs.GetHairs(3).size()) {
+						extraHairTest = 1;
+						return;
+					}
```

Comment:
get rid of the duplicate code. Pick the character you're working with at the start and then do the things to it